### PR TITLE
feat: add `Get-vRLSCMProductPassword`

### PR DESCRIPTION
**Pull request by burnsjared0415**
### Summary

Adds ` Get-vRSLCMProductPassword ` to be consumed in downstream in `VMware.CloudFoundation.PowerManagement` to get the decrypted password from the Locker for the purposes of authentication.

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
...
+       
+           
+        $uri = "https://$vrslcmAppliance/lcm/locker/api/v2/passwords/$lockervmid/decrypted"
+        $jsonSpec = @()
+        $jsonSpec += [pscustomobject]@{
+        'rootPassword' = $vrslcmrootpass
+        }
+        $body = $jsonSpec | ConvertTo-Json -Depth 50
+        $response = Invoke-RestMethod $uri -Method 'POST' -Headers $vrslcmHeaders -Body $body
+        $response
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:37780]: Trying to not have an extra variable here for newer functions as its overkill

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
...
+ Function Get-vRSLCMProductPassword {
+    <#
+        .SYNOPSIS
+        Get Product Password in VMware Aria Suite Lifecycle
+
+        .DESCRIPTION
+        The Get-vRSLCMProductPassword cmdlet gets the product password for a certain product in VMware Aria Suite Lifecycle
+
+        .EXAMPLE
+        Get-vRSLCMProductPassword -lockervmid 8de0ae90-b2ac-48b8-a5c3-d16e3254298b -vrslcmrootpass VMw@re1!
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:37755]:
```suggestion
        Get-vRSLCMProductPassword -lockerVmid 8de0ae90-b2ac-48b8-a5c3-d16e3254298b -vrslcmRootPass VMw@re1!
```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
...
+
+        .PARAMETER lockervmid
+        The locker VMID of the product to get the password for.
+
+        .PARAMETER vrslcmrootpass
+        The root password of the VMware Aria Suite Lifecycle appliance.   
+    #>
+
+    Param (
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$lockervmid,
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:37766]:
```suggestion
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$lockerVmid,
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
...
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vrslcmrootpass
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:37767]:
```suggestion
        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$vrslcmRootPass
```

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
 } 
 Export-ModuleMember -Function Get-vRSLCMEnvironmentVMs
 
+ Function Get-vRSLCMProductPassword {
+    <#
+        .SYNOPSIS
+        Get Product Password in VMware Aria Suite Lifecycle
```
Comment (GaryJBlake) [inline PowerValidatedSolutions.psm1:37749]:
```suggestion
        Get Product Password from VMware Aria Suite Lifecycle
```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# Get-vRSLCMProductPassword
+
+## Synopsis
+
+Get Product Password in VMware Aria Suite Lifecycle
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md:5]:
```suggestion
Get Product Password from VMware Aria Suite Lifecycle

```

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+The `Get-vRSLCMProductPassword` cmdlet gets the product password for a certain product in VMware Aria Suite Lifecycle
+
+## Examples
+
+### Example 1
+
+```powershell
+Get-vRSLCMProductPassword -lockervmid 8de0ae90-b2ac-48b8-a5c3-d16e3254298b -vrslcmrootpass VMw@re1!
+This example gets all environments in VMware Aria Suite Lifecycle
```
Comment (GaryJBlake) [inline docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md:23]:
```suggestion
This example retrieves the product password based on the vmid provided

```

**Review by GaryJBlake — CHANGES_REQUESTED** [review]

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md (hunk 7 above)
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md:23]: Resolved in cd4ef29.

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md (hunk 6 above)
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md:5]: Resolved in cd4ef29.

File: PowerValidatedSolutions.psm1 (hunk 5 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:37749]: Resolved in cd4ef29.

File: PowerValidatedSolutions.psm1 (hunk 3 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:37766]: Resolved in cd4ef29.

File: PowerValidatedSolutions.psm1 (hunk 2 above)
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:37755]: Resolved in cd4ef29.

File: PowerValidatedSolutions.psm1 [hunk 8]
```diff
@@ -37743,6 +37743,47 @@ Function Get-vRSLCMEnvironmentVMs {
...
+
+        .DESCRIPTION
+        The Get-vRSLCMProductPassword cmdlet gets the product password for a certain product in VMware Aria Suite Lifecycle
+
+        .EXAMPLE
+        Get-vRSLCMProductPassword -lockervmid 8de0ae90-b2ac-48b8-a5c3-d16e3254298b -vrslcmrootpass VMw@re1!
+        This example gets all environments in VMware Aria Suite Lifecycle
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1:37756]: Resolved in cd4ef29.

File: PowerValidatedSolutions.psm1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply PowerValidatedSolutions.psm1]: Resolved in cd4ef29.

File: docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+```
```
[the remark this answers is not in the document]
Comment (tenthirtyam) [reply docs/documentation/functions/aria-suite/aria-suite-lifecycle/Get-vRSLCMProductPassword.md]: Resolved in ad1b306.

**Review by tenthirtyam — APPROVED** [review]
LGTM!

**Pull request merged**